src/sp_utils.h

```diff
@@ -71,6 +71,7 @@ extern int sp_debug_stderr;
 #define GET_SUFFIX(x) (x == 1) ? "st" : ((x == 2) ? "nd" : "th")
 
 const char *get_ipaddr(void);
+__attribute__((__format__(printf, 4, 5)))
```

Comment:
Adding `ifdef` guards would make it a tad more portable:

```C
#if defined __has_attribute
#  if __has_attribute (__format__)
#    __attribute__((__format__(printf, 4, 5)))
#  endif
#endif
```

Reply:
done